apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts

```diff
@@ -125,57 +126,58 @@ async function calculateTokensStats(
 }
 
 export async function calculatePoolStats({
-  round,
+  startAt,
+  endAt,
   poolId,
 }: {
-  round: Round;
+  startAt: Date;
+  endAt: Date;
   poolId: string;
 }): Promise<calculatePoolData> {
   const pool = new Pool(poolId);
   const network = String(pool.network ?? 1);
-
+  const startAtUnixtimestamp = startAt.getTime() / 1000;
+  const endAtUnixtimestamp = endAt.getTime() / 1000;
   const [
     balPriceUSD,
     [tvl, volume, symbol, tokenBalance],
     votingShare,
     [feeAPR, collectedFeesUSD],
     tokensAPR,
   ] = await Promise.all([
-    getBALPriceByRound(round.startDate, round.endDate),
-    fetchPoolAveragesInRange(
+    getBALPriceForDateRange(startAt, endAt),
+    fetchPoolAveragesForDateRange(
       poolId,
       network,
-      round.startDate.getTime() / 1000,
-      round.endDate.getTime() / 1000,
+      startAtUnixtimestamp,
+      endAtUnixtimestamp,
     ),
-    getPoolRelativeWeight(poolId, round.endDate.getTime() / 1000),
-    getFeeApr(
+    getPoolRelativeWeight(poolId, endAtUnixtimestamp),
+    getFeeAprForDateRange(
       poolId,
       network,
-      round.startDate.getTime() / 1000,
-      round.endDate.getTime() / 1000,
+      startAtUnixtimestamp,
+      endAtUnixtimestamp,
     ),
     //TODO: on #BAL-795 use another strategy for cache using the poolId
-    getPoolTokensAprForDate(
+    getPoolTokensAprForDateRange(
       network,
       poolId as Address,
-      //Currently, this is calculating the APR on the last day of the round.
-      //This should be changed on #BAL-799
-      round.activeRound
-        ? Math.round(new Date().getTime() / 1000)
-        : round.endDate.getTime() / 1000,
+      startAtUnixtimestamp,
+      endAtUnixtimestamp,
     ),
   ]);
 
   const tokens = await calculateTokensStats(
-    round,
+    endAt,
     pool.tokens,
     network,
     tokenBalance,
   );
```

Comment:
🤔 wouldn't we also need startAt here?

Reply:
but this way we're only outputting the same token stats for every day, no?

Reply:
No since we fetch it like this for bal:
`https://coins.llama.fi/prices/historical/<timestamp>/ethereum:0xba100000625a3754423978a60c9317c58a424e3d?searchWidth=6h`

[22nd](https://coins.llama.fi/prices/historical/1695340800/ethereum:0xba100000625a3754423978a60c9317c58a424e3d?searchWidth=6h) - 3.14
[21rst](https://coins.llama.fi/prices/historical/1695254400/ethereum:0xba100000625a3754423978a60c9317c58a424e3d?searchWidth=6h) - 3.23
[20th](https://coins.llama.fi/prices/historical/1695168000/ethereum:0xba100000625a3754423978a60c9317c58a424e3d?searchWidth=6h) - 3.28